packages/gatsby-react-router-scroll/src/ScrollBehaviorContext.js

```diff
@@ -95,11 +87,20 @@ class ScrollContext extends React.Component {
   }
 
   render() {
-    return React.Children.only(this.props.children)
+    return (
+      <ScrollBehaviorContext.Provider
+        value={{
+          registerElement: this.registerElement,
+          unregisterElement: this.unregisterElement,
+          scrollBehavior: this.scrollBehavior,
+        }}
```

Comment:
Wouldn't this re-render child components whenever this component re-renders?
_because a new object instance passed on each render._

Reply:
I think you're right. Let me change to use `this`